cpg_infra/abstraction/group_settings.py

```diff
@@ -0,0 +1,113 @@
+"""
+Contains pulumi.dynamic.ResourceProvider implementations
+for Google Groups settings.
+"""
+
+from functools import cache
+import pulumi
+import pulumi.dynamic
+import google.auth
+import googleapiclient.discovery
+from google.auth.transport.requests import Request
+
+
+class GroupSettings(pulumi.dynamic.Resource):
+    """A Pulumi dynamic resource for Google Groups settings."""
+
+    group_email: pulumi.Output[str]
+    settings: pulumi.Output[dict]
+
+    def __init__(
+        self,
+        name: str,
+        group_email: str,
+        settings: dict,
+        opts: pulumi.ResourceOptions | None = None,
+    ):
+        super().__init__(
+            GroupSettingsProvider(),
+            name,
+            {
+                'group_email': pulumi.Output.from_input(group_email),
+                'settings': pulumi.Output.from_input(settings),
+            },
+            opts,
+        )
+
+
+class GroupSettingsProvider(pulumi.dynamic.ResourceProvider):
+    """A Pulumi dynamic resource provider for Google Groups settings."""
+
+    def create(self, props):
+        """Creates a Google Groups settings resource."""
+        group_email = props['group_email']
+        settings = props['settings']
+        updated_settings = update_group_settings(group_email, settings)
+        # The response contains *all* settings, so subset to relevant keys.
+        updated_settings = {k: updated_settings[k] for k in settings}
+        outputs = {
+            'group_email': group_email,
+            'settings': updated_settings,
+        }
+        return pulumi.dynamic.CreateResult(
+            id_=f'google_group_settings::{group_email}', outs=outputs
+        )
+
+    def read(self, id_, props):
+        """Reads a Google Groups settings resource."""
+        group_email = props['group_email']
+        settings = props['settings']
+        current_settings = get_group_settings(group_email)
+        # The response contains *all* settings, so subset to relevant keys.
+        current_settings = {k: current_settings[k] for k in settings}
```

Comment:
Should this be a `.get` in case a key suddenly ends up missing, which would then cause an update rather than an error?